content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,295 @@
+
+## A fragmented future
+
+With the division of responsibility described above, there are many points in the pipeline that may impose limitations in functionality. The graphic system might not provide the higher-level API to use functionality in the graphics engine, or a graphic device might not provide support for instructions coming in from the graphics engine. While this has mainly been a hypothetical situation prior to R 4.1, it is now the new reality. As users, this is most apparent in the choice of graphic device. After all, you don't expect the graphic system to be capable of something outside of its API, simply because new features were announced for the graphics device. However, if a graphic device lack support you may end up being surprised at what it renders.
+
+When it comes to graphic systems you can expect that grid will be the first, perhaps only, system that ends up supporting new features in the graphics engine. Part of the reason for that is that the grid API is more powerful in general and as new and more complex graphic powers are exposed it can be easiest to make them fit into the most expressive API. This is definitely the case for the latest batch of new features, but I also expect it to be the case going forward. Just because a functionality is exposed in grid, doesn't mean that it can easily be handled in e.g. ggplot2. I'll address what the new features may mean for the future of ggplot2 in the end of the post.
+
+For graphic devices the water is more muddled. Not all devices are under active development and those devices are unlikely to add support for new features. Further, it may be that a graphics device writes to a format, or uses a library, that do not support a new feature provided by the graphics engine. The bottom line is that we can expect an increased fragmentation of the graphics devices in R in terms of which will be up to specs with the latest graphics engine features. It appears that the cairo based devices along with the pdf device from grDevices can be expected to stay current. On our end, we will do our best to make sure that our graphic device packages (currently ragg and svglite) will stay on top of any new additions to the graphics engine.
+
+## The new features
+
+OK, so we've talked a lot about some new features without ever going into details with what they are. If you've never felt constrained by the capabilities of the graphics in R you may be forgiven for thinking that this is all a big fuzz over nothing. You may be right, but new capabilities will often allow the ecosystem to evolve in new and unexpected ways to the benefit of all.
```

Comment:
Maybe it would be better to start with the new features, and then go in to the details of what changes were needed to make them possible?

Reply:
I feel like the amount of talking I'm doing before diving into new features is pretty mild, so I'd rather have it here instead of an afterthought in the end... but that is just my take

Reply:
Maybe if you just previewed the new features in the first paragraph?

> The release of ragg 1.2 and svglite 2.1 brings support for some exciting new graphics engine features, including gradients and patterns, that were added to R 4.1 by Paul Murrell from R Core.

Also just noticed you said "fuzz" instead of "fuss".